force-app/main/default/lwc/apexWireMethodWithParams/__tests__/apexWireMethodWithParams.test.js

```diff
@@ -0,0 +1,85 @@
+import { createElement } from 'lwc';
+import ApexWireMethodWithParams from 'c/apexWireMethodWithParams';
+import { registerLdsTestWireAdapter } from '@salesforce/wire-service-jest-util';
+import findContacts from '@salesforce/apex/ContactController.findContacts';
+
+// Realistic data with a list of contacts
+const mockFindContacts = require('./data/findContacts.json');
+// An empty list of records to verify the component does something reasonable
+// when there is no data to display
+const mockFindContactsNoRecords = require('./data/findContactsNoRecords.json');
+
+// Register as an LDS wire adapter. Some tests verify that provisioned values trigger desired behavior.
+const findContactsAdapter = registerLdsTestWireAdapter(findContacts);
+
+describe('c-apex-wire-method-with-params', () => {
+    beforeAll(() => {
+        jest.useFakeTimers();
+    });
+
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    describe('findContacts @wire data', () => {
+        it('with one record', () => {
+            const element = createElement('c-apex-wire-method-with-params', {
+                is: ApexWireMethodWithParams
+            });
+            document.body.appendChild(element);
+
+            const inputEl = element.shadowRoot.querySelector('lightning-input');
+            inputEl.value = 'Amy';
+            inputEl.dispatchEvent(new CustomEvent('change'));
```

Comment:
Similar to my comment in apexImperativeMethodWithParams.test.js, this user input does not affect the outcome of the test since we emit the same set of test data regardless of input.

It'd be great to have a test that verifies this input is passed along to the `@wire`. Try the `findContactsAdapter.getLastConfig()` API.

Reply:
Changed.